### PR TITLE
Ability to sell towers

### LINKED ISSUE #64: Option to sell tower
This came up in play test with Fred. Player may want to remove a tower.

Currently there's no way for a tower to be destroyed either, so they're permanent. Need to discuss if there would be a refund and how much.

### PR DESCRIPTION
**Pull request by rockokw**
Users can now sell towers built during a previous build phase. Refund is set at half the purchase price.

I'll incorporate this into the undo chain as part of #66.

Closes #64 

image

### CODE REVIEW

File: frontend/src/entities/towers/Tower.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
 import Entity from "../Entity";
 
+export const Status = {
+    PENDING: Symbol("pending"),
+    BUILT: Symbol("built"),
+};
```
Comment (cb299792458) [inline frontend/src/entities/towers/Tower.js:6]: Haven't seen symbols before, are they basically like enums, but useable in js?

Comment (cb299792458) [reply frontend/src/entities/towers/Tower.js:6]: (At least in this use case)

File: frontend/src/components/ui/SellMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        { x: 0, y: 0, z: -1.25 }
+    );
+
+    const sellTower = () => {
+        game.towers[selectedTower.x][selectedTower.y] = undefined;
+        gameMap.removeTower(selectedTower.x, selectedTower.y);
+
+        const t = TOWERS.get(selectedTower.name);
+        game.gold += 0.5 * t.price;
+        setSelectedTower(null);
```
Comment (cb299792458) [inline frontend/src/components/ui/SellMenu.jsx:64]:
```suggestion
        reset();
```

File: frontend/src/components/ui/SellMenu.jsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+const RADIUS = 0.2;
+const HEIGHT = 0.4;
+const RADIAL_SEGMENTS = 6;
+
+export default function SellMenu({ game }) {
+    const { gameMap, mouseInput } = game;
+    const { width, depth } = gameMap;
+    const [selectedTower, setSelectedTower] = useState(null);
+
+    const reset = () => {
```
Comment (cb299792458) [inline frontend/src/components/ui/SellMenu.jsx:21]: Not a dealbreaker, but it's a pretty generic function name and doesn't clean up much vs just calling setSelectedTower()

**Review by cb299792458 — APPROVED** [review]

File: frontend/src/entities/towers/Tower.js (hunk 1 above)
Comment (rockokw) [reply frontend/src/entities/towers/Tower.js:6]: Yea, I did this to basically have it behave like an enum. `Symbol()`, as I've used it, returns globally unique values for each call. It's a bit odd as some things are in js but should get the job done until we transition to Typescript.

File: frontend/src/components/ui/SellMenu.jsx (hunk 3 above)
Comment (rockokw) [reply frontend/src/components/ui/SellMenu.jsx:21]: I was anticipating more in there but luckily things turned out simpler. I'll just get rid of it.

**Pull request merged**